engine-tests/src/test_utils/mocked_external.rs

```diff
@@ -0,0 +1,176 @@
+use near_vm_logic::mocks::mock_external::MockedExternal;
+
+pub(crate) struct MockedExternalWithTrie<'a> {
+    underlying: &'a mut MockedExternal,
```

Comment:
Is there any particular motivation to use a reference here and build this object from the Mocked External all the time, instead of storing in the AuroraRunner object the MockedExternalWithTrie from the beginning (without requiring a reference to the `underlying` object).

I see the only place we are using `self.ext` without building the `MockedExternalWithTrie` is in the one_shot scenario, and even probably there it makes sense to use MockedExternalWithTrie as well.

Reply:
I thought about having `MockedExternalWithTrie` own the underlying `MockedExternal`. Using a reference seemed like a more flexible approach, but you're right it probably just adds more complexity and we will always want to use this with `AuroraRunner` (and the derived one-shot runner). I will change it as you have suggested.